samza-core/src/main/java/org/apache/samza/util/ThreadUtil.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.util;
+
+import java.lang.management.LockInfo;
+import java.lang.management.ManagementFactory;
+import java.lang.management.MonitorInfo;
+import java.lang.management.ThreadInfo;
+import java.lang.management.ThreadMXBean;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ThreadUtil {
+  private static final Logger LOGGER = LoggerFactory.getLogger(ThreadUtil.class);
+  private static final ThreadMXBean THREAD_MX_BEAN = ManagementFactory.getThreadMXBean();
+
+  public static void logThreadDump(String message) {
+    try {
+      ThreadInfo[] threadInfo = THREAD_MX_BEAN.dumpAllThreads(true, true);
+      StringBuilder sb = new StringBuilder();
+      sb.append(message).append("\n");
+      for (ThreadInfo ti: threadInfo) {
+        sb.append(toString(ti)).append("\n");
+      }
+      System.out.println(sb.toString());
```

Comment:
It looks like the previous impl used log4j to log. Did you intend to just print to stdout now?

Reply:
Oh, this was for testing, will fix.